java/arcs/core/storage/handle/HandleManager.kt

```diff
@@ -0,0 +1,133 @@
+
+typealias SingletonData<T> = CrdtSingleton.Data<T>
+typealias SingletonOp<T> = CrdtSingleton.IOperation<T>
+typealias SingletonStoreOptions<T> = StoreOptions<SingletonData<T>, SingletonOp<T>, T?>
+typealias SingletonStore<T> = Store<SingletonData<T>, SingletonOp<T>, T?>
+typealias SingletonHandle<T> = SingletonImpl<T>
+typealias SingletonActivationFactory<T> = ActivationFactory<SingletonData<T>, SingletonOp<T>, T?>
+
+typealias SetData<T> = CrdtSet.Data<T>
+typealias SetOp<T> = CrdtSet.IOperation<T>
+typealias SetStoreOptions<T> = StoreOptions<SetData<T>, SetOp<T>, Set<T>>
+typealias SetStore<T> = Store<SetData<T>, SetOp<T>, Set<T>>
+typealias SetHandle<T> = CollectionImpl<T>
+typealias SetActivationFactory<T> = ActivationFactory<SetData<T>, SetOp<T>, Set<T>>
+
+interface ActivationFactoryFactory {
+    fun singletonFactory(): SingletonActivationFactory<RawEntity>
+    fun setFactory(): SetActivationFactory<RawEntity>
+}
+
+/**
+ * [HandleManager] is a convenience for creating handles using a provided store factory.
+ *
+ * Handles that are used by end-users will deal with [RawEntity], so this helper only bothers to
+ * create those types.
+ *
+ * It will create a [StorageProxy] for each new storage key, and keep a reference to it for as
+ * long as the [HandleFactory] exists.
+ *
+ * Instantiate it with the context and lifecycle that should own the resulting activate stores.
+ */
+class HandleManager(private val aff: ActivationFactoryFactory? = null) {
+    companion object {
+        /**
+         * Convenience for making a ramdisk-backed reference mode key
+         */
+        fun ramdiskStorageKeyForName(name: String) = ReferenceModeStorageKey(
+            backingKey = RamDiskStorageKey("$name-backing"),
+            storageKey = RamDiskStorageKey("$name-storage")
+        )
+    }
+
+    private val singletonProxies = mutableMapOf<StorageKey, SingletonProxy<RawEntity>>()
+    private val setProxies = mutableMapOf<StorageKey, SetProxy<RawEntity>>()
+
+    private val singletonStores = mutableMapOf<StorageKey, SingletonStore<RawEntity>>()
+    private val setStores = mutableMapOf<StorageKey, SetStore<RawEntity>>()
+
+    /**
+     * Create a new SingletonHandle backed by an Android [ServiceStore]
+     */
+    suspend fun singletonHandle(
+        storageKey: StorageKey,
+        schema: Schema,
+        callbacks: Callbacks<SingletonOp<RawEntity>>? = null
+    ): SingletonHandle<RawEntity> {
+        val storeOptions = SingletonStoreOptions<RawEntity>(
+            storageKey = storageKey,
+            existenceCriteria = ExistenceCriteria.MayExist,
+            type = SingletonType(EntityType(schema)),
+            mode = StorageMode.ReferenceMode
+        )
+
+        val store = singletonStores.getOrPut(storageKey) {
+            Store(storeOptions)
+        }
+
+        val activeStore = aff?.let {
+            store.activate(it.singletonFactory())
+        } ?: store.activate()
```

Comment:
I think `activate()` allows for a null argument, right? If so, this could be simplified to:

```kotlin
val activeStore = store.activate(aff?.singletonFactory())
```